app/frontend/src/features/donations/DonationsBox.tsx

```diff
@@ -0,0 +1,270 @@
+  },
+}));
+
+export default function Donations() {
+  const classes = useStyles();
+
+  return (
+    <section className={classes.donationsBox}>
+      <Typography className={classes.marginBottom2} variant="h3">
+        {DONATIONSBOX_TITLE}
+      </Typography>
+      <div className={classes.donationsBoxRow}>
+        <Button
+          classes={{
+            root: classes.buttonSecondary,
+            label: classes.buttonSecondaryText,
+          }}
+        >
+          {DONATIONSBOX_MONTHLY}
+        </Button>
+        <Button
+          classes={{
+            root: classes.buttonSecondaryActive,
+            label: classes.buttonSecondaryTextActive,
+          }}
+        >
+          {DONATIONSBOX_ONETIME}
+        </Button>
+      </div>
+
+      <Divider className={classes.marginY2} />
+
+      <div className={classes.donationsBoxRow}>
+        <div className={classes.donationsBoxSubRow}>
+          {Array.from(DONATIONSBOX_VALUES?.values() ?? [])
+            .slice(0, 2)
+            .map((value) =>
+              value ? (
+                <Button
+                  classes={{
+                    root: classes.buttonSecondary,
+                    label: classes.buttonSecondaryText,
+                  }}
+                >
+                  {`${value.currency}${value.amount}`}
+                </Button>
+              ) : null
+            )}
+        </div>
+        <div className={classes.donationsBoxSubRow}>
+          {Array.from(DONATIONSBOX_VALUES?.values() ?? [])
+            .slice(2, 4)
+            .map((value) =>
+              value ? (
+                <Button
+                  classes={{
+                    root: classes.buttonSecondary,
+                    label: classes.buttonSecondaryText,
+                  }}
+                >
+                  {`${value.currency}${value.amount}`}
+                </Button>
+              ) : null
+            )}
+        </div>
+        <div className={classes.donationsBoxSubRow}>
+          {Array.from(DONATIONSBOX_VALUES?.values() ?? [])
+            .slice(4, 6)
+            .map((value) =>
+              value ? (
+                <Button
+                  classes={{
+                    root: classes.buttonSecondary,
+                    label: classes.buttonSecondaryText,
+                  }}
+                >
+                  {`${value.currency}${value.amount}`}
+                </Button>
+              ) : null
+            )}
+        </div>
+        <div className={classes.donationsBoxSubRow}>
+          {Array.from(DONATIONSBOX_VALUES?.values() ?? [])
+            .slice(6, 7)
+            .map((value) =>
+              value ? (
+                <Button
+                  classes={{
+                    root: classes.buttonSecondary,
+                    label: classes.buttonSecondaryText,
+                  }}
+                >
+                  {`${value.currency}${value.amount}`}
+                </Button>
+              ) : null
+            )}
```

Comment:
I think it's better to inline and repeat the `Button` JSX markup here instead. This array mapping thing is not really adding any value and makes the code more obscure to read imo, as you are hardcoding the different currency values by repeating the rows anyway, so this is adding an unnecessary layer of indirection with the JSX being generated from the array.